velox/connectors/hive/tests/HivePartitionUtilTest.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/connectors/hive/HivePartitionUtil.h"
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/vector/tests/utils/VectorTestBase.h"
+
+#include "gtest/gtest.h"
+
+using namespace facebook::velox::connector::hive;
+using namespace facebook::velox;
+
+class HivePartitionUtilTest : public ::testing::Test,
+                              public test::VectorTestBase {
+ protected:
+  template <typename T>
+  VectorPtr makeDictionary(const std::vector<T>& data) {
+    auto base = makeFlatVector(data);
+    auto indices = makeIndices(data.size(), [](auto row) { return row; });
```

Comment:
This dictionary is not interesting. Consider testing constant vectors as well as dictionary vectors where a few values repeat many times.

Reply:
Revised for dictionary vector. Testing on constant vector is included as the last column in the test case.